plugin/trino-hive/src/main/java/io/trino/plugin/hive/orc/OrcPageSource.java

```diff
@@ -147,12 +160,27 @@ public Page getNextPage()
         OptionalLong startRowId = originalFileRowId.isPresent() ?
                 OptionalLong.of(originalFileRowId.get() + recordReader.getFilePosition()) : OptionalLong.empty();
 
+        if (deletedRows.isPresent()) {
+            boolean deletedRowsYielded = !deletedRows.get().loadOrYield();
+            if (deletedRowsYielded) {
+                remainingPage = Optional.of(page);
+                localMemoryContext.setBytes(page.getRetainedSizeInBytes());
+                Page emptyPage = page.getRegion(0, 0);
```

Comment:
@sopel39 is this reasonable way to construct an empty page? Specifically can it have performance/memory implications like unnecessary materialization of lazy blocks?

Reply:
`io.trino.spi.connector.ConnectorPageSource#getNextPage` is allowed to return null (instead of an empty page)

Reply:
Thanks. That makes it easier.